In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

print("Libraries imported successfully!")

Libraries imported successfully!


In [4]:
import pandas as pd

df = pd.read_csv("../data/traffic_dataset.csv")

print("Dataset loaded successfully!")

Dataset loaded successfully!


In [5]:
df = pd.read_csv(r"D:\Traffice-System\data\traffic_dataset.csv")

In [6]:
df.head()

,filename,date,timestamp,direction,day_night,weather,start_frame,number_of_frames,traffic_class,notes
0,cctv052x2004080516x01638,20040805,16.01638,south,day,overcast,2,53,medium,NaN
1,cctv052x2004080516x01639,20040805,16.01639,south,day,overcast,2,53,medium,NaN
2,cctv052x2004080516x01640,20040805,16.01640,south,day,overcast,2,48,light,NaN
3,cctv052x2004080516x01641,20040805,16.01641,south,day,overcast,2,52,medium,NaN
4,cctv052x2004080516x01642,20040805,16.01642,south,day,overcast,2,51,medium,NaN


In [7]:
df.tail()

,filename,date,timestamp,direction,day_night,weather,start_frame,number_of_frames,traffic_class,notes
249,cctv052x2004080619x00104,20040806,19.00104,south,day,clear,2,53,light,NaN
250,cctv052x2004080620x00105,20040806,20.00105,south,day,clear,2,52,light,NaN
251,cctv052x2004080620x00106,20040806,20.00106,south,day,clear,2,53,light,NaN
252,cctv052x2004080620x00107,20040806,20.00107,south,day,clear,2,53,light,NaN
253,cctv052x2004080620x00108,20040806,20.00108,south,day,clear,2,52,light,NaN


In [8]:
print("Number of rows:", df.shape[0])
print("Number of columns:", df.shape[1])

Number of rows: 254
Number of columns: 10


In [9]:
print("Column names:")

for column in df.columns:
    print(column)

Column names:
filename
date
timestamp
direction
day_night
weather
start_frame
number_of_frames
traffic_class
notes


In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 254 entries, 0 to 253
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   filename          254 non-null    str    
 1   date              254 non-null    int64  
 2   timestamp         254 non-null    float64
 3   direction         254 non-null    str    
 4   day_night         254 non-null    str    
 5   weather           254 non-null    str    
 6   start_frame       254 non-null    int64  
 7   number_of_frames  254 non-null    int64  
 8   traffic_class     254 non-null    str    
 9   notes             14 non-null     str    
dtypes: float64(1), int64(3), str(6)
memory usage: 20.0 KB


In [11]:
df.describe()

,date,timestamp,start_frame,number_of_frames
count,2.540000e+02,254.000000,254.0,254.000000
mean,2.004081e+07,13.841694,2.0,52.429134
std,4.280224e-01,4.278468,0.0,1.297953
min,2.004080e+07,6.018200,2.0,43.000000
25%,2.004081e+07,10.018853,2.0,52.000000
50%,2.004081e+07,15.000355,2.0,53.000000
75%,2.004081e+07,17.016627,2.0,53.000000
max,2.004081e+07,20.017000,2.0,53.000000


In [12]:
df.describe(include="all")

,filename,date,timestamp,direction,day_night,weather,start_frame,number_of_frames,traffic_class,notes
count,254,2.540000e+02,254.000000,254,254,254,254.0,254.000000,254,14
unique,254,NaN,NaN,1,1,3,NaN,NaN,3,2
top,cctv052x2004080516x01638,NaN,NaN,south,day,overcast,NaN,NaN,light,drops on lens
freq,1,NaN,NaN,254,254,125,NaN,NaN,165,13
mean,NaN,2.004081e+07,13.841694,NaN,NaN,NaN,2.0,52.429134,NaN,NaN
std,NaN,4.280224e-01,4.278468,NaN,NaN,NaN,0.0,1.297953,NaN,NaN
min,NaN,2.004080e+07,6.018200,NaN,NaN,NaN,2.0,43.000000,NaN,NaN
25%,NaN,2.004081e+07,10.018853,NaN,NaN,NaN,2.0,52.000000,NaN,NaN
50%,NaN,2.004081e+07,15.000355,NaN,NaN,NaN,2.0,53.000000,NaN,NaN
75%,NaN,2.004081e+07,17.016627,NaN,NaN,NaN,2.0,53.000000,NaN,NaN


In [13]:
print("Missing values in each column:")

print(df.isnull().sum())

Missing values in each column:
filename              0
date                  0
timestamp             0
direction             0
day_night             0
weather               0
start_frame           0
number_of_frames      0
traffic_class         0
notes               240
dtype: int64


In [14]:
missing_percentage = (
    df.isnull().sum() / len(df) * 100
)

print(missing_percentage)

filename             0.000000
date                 0.000000
timestamp            0.000000
direction            0.000000
day_night            0.000000
weather              0.000000
start_frame          0.000000
number_of_frames     0.000000
traffic_class        0.000000
notes               94.488189
dtype: float64


In [15]:
print("Number of duplicate rows:")

print(df.duplicated().sum())

Number of duplicate rows:
0


In [16]:
for column in df.columns:
    print("\n", column)
    print(df[column].unique())


 filename
<StringArray>
['cctv052x2004080516x01638', 'cctv052x2004080516x01639',
 'cctv052x2004080516x01640', 'cctv052x2004080516x01641',
 'cctv052x2004080516x01642', 'cctv052x2004080516x01643',
 'cctv052x2004080516x01644', 'cctv052x2004080516x01645',
 'cctv052x2004080516x01646', 'cctv052x2004080516x01647',
 ...
 'cctv052x2004080619x00099', 'cctv052x2004080619x00100',
 'cctv052x2004080619x00101', 'cctv052x2004080619x00102',
 'cctv052x2004080619x00103', 'cctv052x2004080619x00104',
 'cctv052x2004080620x00105', 'cctv052x2004080620x00106',
 'cctv052x2004080620x00107', 'cctv052x2004080620x00108']
Length: 254, dtype: str

 date
[20040805 20040806]

 timestamp
[16.01638 16.01639 16.0164  16.01641 16.01642 16.01643 16.01644 16.01645
 16.01646 16.01647 16.01648 16.01649 16.0165  17.01652 17.01653 17.01654
 17.01655 17.01656 17.01657 17.01658 17.01659 17.0166  17.01661 17.01662
 17.01663 17.01664 17.01665 18.01666 18.01667 18.01668 18.01669 18.0167
 18.01671 18.01672 18.01673 18.01674 18.01675 

In [17]:
print(df.nunique())

filename            254
date                  2
timestamp           254
direction             1
day_night             1
weather               3
start_frame           1
number_of_frames      9
traffic_class         3
notes                 2
dtype: int64


In [18]:
print(df["traffic_class"].value_counts())

traffic_class
light     165
medium     45
heavy      44
Name: count, dtype: int64


In [19]:
class_percentage = (
    df["traffic_class"]
    .value_counts(normalize=True) * 100
)

print(class_percentage.round(2))

traffic_class
light     64.96
medium    17.72
heavy     17.32
Name: proportion, dtype: float64


In [20]:
print(df["weather"].value_counts())

weather
overcast    125
rain         78
clear        51
Name: count, dtype: int64


In [21]:
print(df["day_night"].value_counts())

day_night
day    254
Name: count, dtype: int64


In [22]:
print(df["direction"].value_counts())

direction
south    254
Name: count, dtype: int64


## Final observation
The dataset contains 254 traffic samples with 10 columns.
The target variable is traffic_class.
The traffic classes are Light, Medium, and Heavy.
The dataset contains information such as weather, direction, day/night, date, and frames.
The dataset was successfully loaded and its basic structure was explored.